client/Requester.ts

```diff
@@ -0,0 +1,13 @@
+import { ipcRenderer } from "electron";
+
+export default class Requester {
+  public request<T>(channel: string, argument?: string): Promise<T> {
+    ipcRenderer.send(channel, argument);
+
+    return new Promise((resolve) => {
+      ipcRenderer.on(channel, (event, response) => {
+        resolve(response);

```

Comment:
So these block until a message is received. Very cool. This behavior is popular in queues (named `ack` or `acknowledge`).
https://stackoverflow.com/questions/28794123/ack-or-nack-in-rabbitmq